src/charm.py

```diff
@@ -184,31 +261,29 @@ def _update(self, event):
         env = create_env(self.model.name, self.app.name, self.config, self._state)
         self.model.unit.set_ports(INTERNAL_API_PORT, CONNECTOR_BUILDER_SERVER_API_PORT)
 
-        for service in list(CONTAINERS.keys()):
-            container = self.unit.get_container(service)
+        for container_name in list(CONTAINERS.keys()):
```

Comment:
So do the services directly map to individual containers?

Reply:
Yes, each of these maps to an individual container.